lottie-compose/src/main/java/com/airbnb/lottie/compose/LottieAnimatable.kt

```diff
@@ -0,0 +1,261 @@
+     *                            recent animation. When [animate] is called with continueFromPreviousAnimate
+     *                            set to true, a delta will be calculated from the most recent [animate] call
+     *                            to ensure that the original progress is unaffected by [snapTo] calls in the
+     *                            middle.
+     *                            Defaults to false if progress is not being snapped to.
+     *                            Defaults to true if progress is being snapped to.
+     */
+    suspend fun snapTo(
+        composition: LottieComposition? = this.composition,
+        progress: Float = this.progress,
+        iteration: Int = this.iteration,
+        resetLastFrameNanos: Boolean = progress != this.progress,
+    )
+
+    /**
+     * Animate a [LottieComposition].
+     *
+     * @param composition The [LottieComposition] that should be rendered.
+     * @param continueFromPreviousAnimate When set to true, this animation will be considered continuous from any
+     *                                    previous animate calls. When set to true 1) parameters will carry over from
+     *                                    their previous value instead of being set to their defaults 2) instead of
+     *                                    starting at the minimum progress, the initial progress will be advanced in
+     *                                    accordance to the amount of time that has passed since the last frame
+     *                                    was rendered.
+     * @param iteration The iteration to start the animation at. Defaults to 1 and starts at 1.
+     * @param iterations The number of iterations to continue running for. Set to 1 to play one time
+     *                   set to [LottieConstants.IterateForever] to iterate forever. Can be set to arbitrary
+     *                   numbers.
+     * @param speed The speed at which the composition should be animated. Can be negative. Defaults to 1.
+     * @param clipSpec An optional [LottieClipSpec] to trim the playback of the composition between two values.
+     * @param initialProgress An optional progress value that the animation should start at. Defaults to the
+     *                        starting progress as defined by the clipSpec and speed. Can be used to resume
+     *                        animations from arbitrary points.
+     * @param cancellationBehavior The behavior that this animation should have when cancelled. In most cases,
+     *                             you will want it to cancel immediately. However, if you have a state based
+     *                             transition and you want an animation to finish playing before moving on to
+     *                             the next one then you may want to set this to [LottieCancellationBehavior.OnFinish].
+     */
+    suspend fun animate(
```

Comment:
@doris4ly. I actually had this in some of the iterations on this branch. One of them copied over the values if `continueFromPreviousAnimate` was true. I could see this going either way and am open to a) default values, b) retained values c) retained of continue is set to true.

Reply:
The best way to work this out is to write some user code for a couple of real-world use cases, and see which option requires the least amount of effort. This comment was just about ergonomics.

Reply:
@doris4lt that's that's the purpose of the Examples pages, actually. This version is a little more verbose but I made the change because in one of my examples, it wound up doing the wrong thing. I'll play around with this a little more though :smile: